### ISSUE TITLE
 `save_in_address_book` has no impact on Address Book

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/graphql-ce#509]
### Steps to reproduce (*)

1. Create customer without address from Storefront
2. Generate Customer Token
3. Create empty cart
4. Add simple product
5. Set address on cart but use `save_in_address_book: true`
```graphql
mutation {
  setShippingAddressesOnCart(
    input: {
      cart_id: "{{ YOUR VALUE }}"
      shipping_addresses: {
        address: {
          firstname: "John"
          lastname: "Doubledo"
          street: "6161 West Centinela Avenue"
          telephone: "+15555555555"
          country_code: "us"
          save_in_address_book: true
          city: "Culver city"
        }
      }
    }
  ) {
    cart {
      shipping_addresses {
        address_id
        
        selected_shipping_method {
          carrier_code
          method_code
          label
          amount
        }
        available_shipping_methods {
          carrier_code
          method_code
          price_incl_tax
          base_amount
          price_excl_tax
          amount
        }
      }
    }
  }
}
```
6. Verify that Customer address was saved and can be used on Storefront

### Expected result (*)

Address is in Address book

### Actual result (*)

No Address is in Address book

Also,  `save_in_address_book` field should be NOT required  (according to
https://github.com/magento/graphql-ce/issues/583)

**Comment by XxXgeoXxX** [comment]
Hi, @TomashKhamlai 
I am not sure that this behavior is a bug, because I saw saved address in "quote_address" table.
After check on frontend - address saved in address book when order was placed.
Maybe we need some clarification about your steps and about default behaviour

**Comment by TomashKhamlai** [comment]
@XxXgeoXxX, were you able to set `region_id` in Step 5? #592

I tried this way:
```graphql
mutation {
  setShippingAddressesOnCart(
    input: {
      cart_id: "kyBqpZRDTTyiOLQTwndJGpqi37vhoWpO"
      shipping_addresses: {
        address: {
          firstname: "John"
          lastname: "Doubledo"
          street: "6161 West Centinela Avenue"
          telephone: "+15555555555"
          postcode: "90230"
          region: "12"
          country_code: "us"
          save_in_address_book: true
          city: "Culver city"
        }
      }
    }
  ) {
    cart {
      shipping_addresses {
        address_id
        postcode
        region {
          
          code
          label
        }
        
        selected_shipping_method {
          carrier_code
          method_code
          label
          amount
        }
        available_shipping_methods {
          carrier_code
          method_code
          price_incl_tax
          base_amount
          price_excl_tax
          amount
        }
      }
    }
  }
}
```


address_id | quote_id | created_at | updated_at | customer_id | save_in_address_book | customer_address_id | address_type | email | prefix | firstname | middlename | lastname | suffix | company | street | city | region | region_id | postcode | country_id | telephone | fax | same_as_billing | collect_shipping_rates | shipping_method | shipping_description | weight | subtotal | base_subtotal | subtotal_with_discount | base_subtotal_with_discount | tax_amount | base_tax_amount | shipping_amount | base_shipping_amount | shipping_tax_amount | base_shipping_tax_amount | discount_amount | base_discount_amount | grand_total | base_grand_total | customer_notes | applied_taxes | discount_description | shipping_discount_amount | base_shipping_discount_amount | subtotal_incl_tax | base_subtotal_total_incl_tax | discount_tax_compensation_amount | base_discount_tax_compensation_amount | shipping_discount_tax_compensation_amount | base_shipping_discount_tax_compensation_amnt | shipping_incl_tax | base_shipping_incl_tax | vat_id | vat_is_valid | vat_request_id | vat_request_date | vat_request_success | validated_country_code | validated_vat_number | gift_message_id | free_shipping
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
189 | 47 | 2019-04-15 17:27:20 | NULL | 3 | 1 | NULL | shipping | yhiteca@gmail.com | NULL | John | NULL | Doubledo | NULL | NULL | 6161 West Centinela Avenue | Culver city | 12 | NULL | 90230 | us | +15555555555 | NULL | 0 | 1 | flatrate_flatrate | Flat Rate - Fixed | 1.0000 | 10.0000 | 10.0000 | 10.0000 | 10.0000 | 0.0000 | 0.0000 | 5.0000 | 5.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 15.0000 | 15.0000 | NULL | [] | NULL | 0.0000 | 0.0000 | 10.0000 | 10.0000 | 0.0000 | 0.0000 | 0.0000 | NULL | 5.0000 | 5.0000 | NULL | NULL | NULL | NULL | NULL | NULL | NULL | NULL | 0


But actually there is no Customer with id `3` in `customer_address_entity`. Can you confirm it?

**Comment by naydav** [comment]
Confirm that bug is present.
We don't have any logic for processing `save_in_address_book` input

Also, need to create the corresponding test
